pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,237 @@
+                self._raise_name_warning(node, node_type, name)
+                current_state = False
+
+        node._is_ascii_only = current_state  # pylint: disable=protected-access
+
+    def _recursive_check_names(self, args: Iterable[nodes.AssignName]):
+        """check names in a possibly recursive list <arg>"""
+        for arg in args:
+            if isinstance(arg, nodes.AssignName):
+                self._check_name("argument", arg.name, arg)
+            else:
+                # pylint: disable-next=fixme
+                # TODO: Check if we can remove this branch because of
+                #       the up to date astroid version used
+                self._recursive_check_names(arg.elts)
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_module(self, node: nodes.Module) -> None:
+        # pylint: disable-next=fixme
+        # TODO check how the node name is defined when calling the real checker
+        #      in the test using the astroid manager the node.name attribute corresponds
+        #      to the filename -> so we always have ".py" as result of the following
+        #      function.
+        #      It would be nice if we could check for an ASCII filename, if not done
+        #      already by other checkers...
+        self._check_name("module", node.name.split(".")[-1], node)
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_functiondef(
+        self, node: Union[nodes.FunctionDef, nodes.AsyncFunctionDef]
+    ) -> None:
+        self._check_name("function", node.name, node)
+
+        # Check argument names
+        arguments = node.args
+
+        # Check position only arguments
+        if arguments.posonlyargs is not None:
+            for pos_only_arg in arguments.posonlyargs:
+                self._check_name("argument", pos_only_arg.name, pos_only_arg)
+
+        # Check "normal" arguments
+        args = arguments.args
+        if args is not None:
+            self._recursive_check_names(args)
+
+        # Check key word only arguments
+        if arguments.kwonlyargs is not None:
+            for kwarg in arguments.kwonlyargs:
+                self._check_name("argument", kwarg.name, kwarg)
+
+    visit_asyncfunctiondef = visit_functiondef
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_global(self, node: nodes.Global) -> None:
+        for name in node.names:
+            self._check_name("const", name, node)
+
+    @pylint.checkers.utils.check_messages("non-ascii-name")
+    def visit_assignname(self, node: nodes.AssignName) -> None:
+        """check module level assigned names"""
+        frame = node.frame()
+        assign_type = node.assign_type()
+        if isinstance(assign_type, nodes.Comprehension):
+            # original inlinevar
+            self._check_name("variable", node.name, node)
+        elif isinstance(frame, nodes.Module):
+            # Here a lot of checks were done in the original module,
```

Comment:
This comment won't make sense in 6 months time, I think we should remove it

Reply:
Removed.